woocommerce/woocommerce-gutenberg-products-block
# Featured Category block: add support for global style

~This PR is blocked by #5515~

This PR adds support for global style for the `Featured Category block`.

Now, the user can edit the style for:
- text-color
- background-color
- font-size
- border-color
- border-radius
- border-width


## What missing:

### Duotone

At this stage, the `Duotone` feature is implemented with the `filter` CSS property. The problem is that on this block, the image is implemented like `background-image`, so we can't use the `filter` property.

### Padding

At this stage, with the current global style API, the padding is applied to the entire wrapper, so I guess that it doesn't make sense enable it.




#4965 

### Screenshots


![image](https://user-images.githubusercontent.com/4463174/148819410-1e86fc49-6c4d-468d-99d8-a1b10baff4ca.png)
*without global style*

![image](https://user-images.githubusercontent.com/4463174/148819524-da366929-cf92-4730-b75e-4ed981b4cc63.png)
*with global style*


### Testing

### Manual Testing

Check out this branch:

1. On `WordPress 5.9`, install and enable the `Gutenberg` plugin.
2. Install and enable the `Twenty Twenty-Two` theme.
3. Add the `Featured Category Block` to a post.
4. On the right sidebar, personalize the styles of the block.
5. Go on the page and check if there are changes.
6. Reset to default using the `Reset` button from the different sections. 
7. Go to Dashboard and select Appearance > Editor (beta). On top of the screen, select Home > Browser all templates > Single Post. When the page is loaded, add the block to the page.
8. On the Editor page click on the `Styles` icon on the right-top corner.
9. Verify that the `Featured Category block` is shown under the `Blocks` section. Personalize again the block.
10. Save your changes.
11. Go on the page created earlier and check if all styles are applied correctly.
12. Edit your previous post/page again.
13. Change again the styles.
14. Save your changes.
15. Check if these styles have priority over the styles from the Site Editor.

### Changelog

> Add support for the global style for the Featured Category block.
